core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -1,15 +1,151 @@
+    // add a 500ms delay for joint-ui.service to fetch the operator metaData
+    // this code is temporary and will be deleted in future PRs when drag
+    // and drop is implemented
+
+    Observable.of([]).delay(500).subscribe(
+      emptyData => {
+        // add some dummy operators and links to show that JointJS works
+        this.graph.addCell(
+          this.jointUIService.getJointjsOperatorElement(
+            'ScanSource',
+            'operator1',
+            100, 100
+          )
+        );
+
+        this.graph.addCell(
+          this.jointUIService.getJointjsOperatorElement(
+            'ViewResults',
+            'operator2',
+            500, 100
+          )
+        );
+
+        const link = this.jointUIService.getJointjsLinkElement(
+          { operatorID: 'operator1', portID: 'out0' },
+          { operatorID: 'operator2', portID: 'in0' }
+        );
+
+        this.graph.addCell(link);
+      }
+    );
+  }
+
+  /**
+   * Creates a JointJS Paper object, which is the JointJS view object responsible for
+   *  rendering the workflow cells and handle UI events.
+   *
+   * JointJS documentation about paper: https://resources.jointjs.com/docs/jointjs/v2.0/joint.html#dia.Paper
+   */
+  private createJointjsPaper(): void {
+
+    const paper = new joint.dia.Paper({
+      // bind the DOM element
+      el: $('#' + this.WORKFLOW_EDITOR_JOINTJS_ID),
+      // bind the jointjs graph model
+      model: this.graph,
+      // set the width and height of the paper to be the width height of the parent wrapper element
+      width: this.getWrapperElementSize().width,
+      height: this.getWrapperElementSize().height,
+      // set grid size to 1px (smallest grid)
+      gridSize: 1,
+      // enable jointjs feature that automatically snaps a link to the closest port when user drops a link
+      snapLinks: true,
+      // disable jointjs default action that can make a link not connect to an operator
+      linkPinning: false,
+      // provide a validation to determine if two ports could be connected (only output connect to input is allowed)
+      validateConnection: validateOperatorConnection,
+      // disable jointjs default action of adding vertexes to the link
+      interactive: { vertexAdd: false },
+      // set a default link element used by jointjs when user creates a link on UI
+      defaultLink: this.jointUIService.getDefaultLinkElement(),
+      // disable jointjs default action that stops propagate click events on jointjs paper
+      preventDefaultBlankAction: false,
+      // disable jointjs default action that prevents normal right click menu showing up on jointjs paper
+      preventContextMenu: false,
+    });
 
-  constructor() { }
+    this.paper = paper;
+  }
 
-  ngOnInit() {
+  /**
+   * get the width and height of the parent wrapper element
+   */
+  private getWrapperElementSize(): {width: number, height: number} {
+    return {
+      width: $('#' + this.WORKFLOW_EDITOR_JOINTJS_WRAPPER_ID).width(),
```

Comment:
I suppose this will give the size of div. But isn't the size of div 0 if there isn't anything inside it? If yes, then how are we using it in createJoinJSPaper to set the size?

Reply:
since the wrapper has the style `texera-workflow-component-body`, which is defined in `workspace.component.scss', the size of the div will be the entire area of the workflow-editor at the center of the browser. 

Since all the sub components wrapper use the class `texera-workflow-component-body`. We defined it in `workspace.component.scss' to avoid redundancy.